lighthouse-core/test/audits/dobetterweb/no-vulnerable-libraries-test.js

```diff
@@ -99,3 +100,41 @@ describe('Avoids front-end JavaScript libraries with known vulnerabilities', ()
     assert.equal(auditResult.score, 1);
   });
 });
+
+// https://github.com/npm/node-semver/issues/166#issuecomment-245990039
+function hasUpperBound(rangeString) {
+  const range = new semver.Range(rangeString);
+  if (!range) return false;
+
+  // For every subset ...
+  for (const subset of range.set) {
+    // Upperbound exists if...
+
+    // < or <= is in one of the subset's clauses (= gets normalized to >= and <).
+    if (subset.some(comparator => comparator.operator.match(/^</))) {
```

Comment:
I'm not sure this entirely works. A `rangeString` of `'*'` ends up throwing here, for instance.

We have an untested homegrown solution vs a tested (by semver) but ugly `gtr` solution...

Reply:
> A rangeString of '*' ends up throwing here, for instance.

Isn't that kinda desired behavior? :)

Reply:
> I'm not sure this entirely works. A rangeString of '*' ends up throwing here, for instance.

Fixed.

> We have an untested homegrown solution vs a tested (by semver) but ugly gtr solution...

Added tests.